micmac_li3ds/api.py

```diff
@@ -74,3 +92,62 @@ def get_sensor_referentials(self, sensor_id):
             err = 'No base referential found for sensor {:d}'.format(sensor_id)
             raise RuntimeError(err)
         return sensor_referentials
+
+
+
+    def get_or_create_object(self, typ, obj, keys, log):
+
+        log.debug('getting or creating {} "{}"'.format(typ, obj))
+
+        if 'id' in obj:
+            # look up by id
+            # raise an error upon lookup failure or value mismatch for specified keys
+            got = self.get_object_by_id(typ, obj['id'])
+            if not got:
+                err = 'Error: {} with id {:d} not in db'.format(typ, obj['id'])
+                raise RuntimeError(err)
+
+            obj_keys = set(obj.keys())
+            got_keys = set(got.keys())
+            keys = obj_keys.intersection(got_keys)
```

Comment:
I don't think you need to create a set object for `got.keys()`. You could do this:

```py
    keys = set(obj.keys()).intersection(got.keys())
```